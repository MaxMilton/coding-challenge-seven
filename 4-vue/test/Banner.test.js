'use strict';

// const { shallowMount } = require('@vue/test-utils');
const { mount } = require('@vue/test-utils');
// const { createRenderer } = require('vue-server-renderer');
const Banner = require('../src/Banner.vue');

describe('Banner component', () => {
  it.skip('prints errors with no props', () => {
    // expect.assertions(1);
    // const spy = jest.spyOn(console, 'error').mockImplementation(() => {});
    // shallowMount(Banner);
    // expect(spy).toHaveBeenCalledTimes(1);
    // spy.mockReset();
    // spy.mockRestore();
  });

  it.skip('renders correctly', () => {
    expect.assertions(1);
    // const renderer = createRenderer();
    // const wrapper = shallowMount(Banner, {
    const wrapper = mount(Banner, {
      // data: {
      propsData: {
        tile: '../tile.jpg',
        title: 'Home And Away',
      },
    });
    // renderer.renderToString(wrapper.vm, (err, str) => {
    //   if (err) throw new Error(err);
    //   expect(str).toMatchSnapshot();
    // });

    // expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
