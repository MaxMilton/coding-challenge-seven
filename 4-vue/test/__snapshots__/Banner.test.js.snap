// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component renders correctly 1`] = `"<!--function (a, b, c, d) { return createElement(vm, a, b, c, d, true); }-->"`;
